storage_service/locations/models/space.py

```diff
@@ -261,6 +261,27 @@ def delete_path(self, delete_path, *args, **kwargs):
         except AttributeError:
             return self._delete_path_local(delete_path)
 
+    def posix_move(self, source_path, destination_path, destination_space, package=None, *args, **kwargs):
+        """
+        Move self.path/source_path direct to destination_space.path/destination_path bypassing staging.
+        """
+        LOGGER.debug('posix_move: source_path: %s', source_path)
+        LOGGER.debug('posix_move: destination_path: %s', destination_path)
+        LOGGER.debug('posix_move: destination_space.path: %s', destination_space.path)
+
+        if not Space.posix_move_supported(self, destination_space):
+            raise ValidationError(_('both spaces must be POSIX filesystems'))
```

Comment:
You can just call the static method on the instance, i.e., `self.posix_move_supported`. It seems odd to call it from the `Space` class here.

Also, the Django `ValidationError` seems odd here to me given that this is not really form validation. Probably better to just use Python's builtin `TypeError`.

Reply:
Also, it seems odd, to duplicate the call to `posix_move_supported` once before you call `posix_move` and once inside it. Why not just attempt to call `.posix_move` and accept the `TypeError` and fallback to `move_to_storage_service`/`move_from_storage_service`, e.g., in resources.py? I can see that the hairiness in package.py may make such an "ask for forgiveness" approach hard (impossible?). If it could be done, I think it would be preferable.